fuel-core/src/schema.rs

```diff
@@ -42,6 +42,19 @@ pub struct Subscription(tx::TxStatusSubscription);
 
 pub type CoreSchema = Schema<Query, Mutation, Subscription>;
 
+fn check_pagination_parameters<T>(
+    first: &Option<i32>,
+    after: &Option<T>,
+    last: &Option<i32>,
+    before: &Option<T>,
+) -> bool {
+    // Condition check will short-circuit so all validation logic is combined
+    first.is_some() && before.is_some()
```

Comment:
What do you think about this variant? It seems more readable to me.

```rust
match (first, before, last, after) {
    (Some(_), Some(_), _, _)
    | (_, _, Some(_), Some(_))
    | (None, Some(_), _, Some(_))
    | (None, None, None, None) => true,
    (_, _, _, _) => false,
}
```

Also, do we need to allow `(None, Some(before), Some(last), None)` and `(Some(first), None, None, Some(after))`?